### PR TITLE
perf: trim package size

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix? (provide a description or issue closes statement)
start changes for #276 

---

### Reproduction steps
1. remove All.json from package (saves ~33MB)
2. Generate array of all categories on init (minus `All` and `i18n`)
2. Remove `['All']` as the default for category and replace with an array of all categories
3. Dedupe category array so we don't dupe items in "this"
4. Minify all but All.json
5. Persist build data across ci jobs

---

### Evidence/screenshot/link to line

You can see my changes in `index.js` and test in `test/index.js`

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[Yes]**
- Does this pr include the updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter through `npm run lint`? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Feature/Maintenance]**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1,46 +1,63 @@
...
+
 beforeEach(() => {
   Items = require('../index')
 })
 afterEach(() => {
   decache('../index.js')
 })
+// after(() => {
+//   console.error(`Average init time: ${(inits.reduce((a, b) => a + b, 0) / inits.length).toFixed(0)}`)
+// })
```
Comment (TobiTenno) [inline test/index.js:25]: commented to allow debugging more later... cause test code, i don't care if it stays in, personally

File: index.js [hunk 2]
```diff
@@ -32,14 +32,25 @@
...
+
 let i18n = {}
 try {
   i18n = require('./data/json/i18n.json')
 } catch (ignored) {
   // can only happen in really weird stuff, and we're already defaulting, so it's ok
 }
-const defaultOptions = { category: ['All'], i18n: false, i18nOnObject: false }
+
+const raw = fs.readdirSync(path.join(__dirname, './data/json/'))
```
Comment (AyAyEm) [inline index.js:46]: raw doesn't mean anything here,  i would suggest `categoryFiles`

File: index.js [hunk 3]
```diff
@@ -32,14 +32,25 @@
...
+const ignored = ['All', 'i18n']
+const all = raw
```
Comment (AyAyEm) [inline index.js:48]: all also doesn't mean much, i would suggest `categories`, `allCategories` or `possibleCategories`

**Review by AyAyEm — CHANGES_REQUESTED** [review]

**Comment by AyAyEm** [comment]
I think this pr is more of a refactor than feature, also we're not changing the size of all.json we're removing it from the package, i think something like `trim all.json from package` or `ignore all.json in package` would fit better.

File: index.js (hunk 2 above)
Comment (TobiTenno) [reply index.js:46]: sure

File: index.js (hunk 3 above)
Comment (TobiTenno) [reply index.js:48]: sure

**Comment by TobiTenno** [comment]
yeah, i actually did that just by allowing specification of spaces

File: build/build.js [hunk 4]
```diff
@@ -105,7 +105,7 @@ class Build {
 
     // All.json (all items in one file)
     all.sort(sort)
-    fs.writeFileSync(path.join(__dirname, '../data/json/All.json'), stringify(all))
+    fs.writeFileSync(path.join(__dirname, '../data/json/All.json'), stringify(all, 2))
```
Comment (AyAyEm) [inline build/build.js:108]: Why would All.json be the single exception, not being minified?

Comment (TobiTenno) [reply build/build.js:108]: because it's the only json:
- not in the build
- used to validate types

Comment (TobiTenno) [reply build/build.js:108]: And validating types is what needs, seemingly, an unminified `All.json` to work correctly. I'm not 100% sure why.

File: build/build.js [hunk 5]
```diff
@@ -100,7 +100,7 @@ class Build {
     for (const category in categories) {
       const data = categories[category].sort(sort)
       all = all.concat(data)
-      fs.writeFileSync(path.join(__dirname, `../data/json/${category}.json`), stringify(data))
+      fs.writeFileSync(path.join(__dirname, `../data/json/${category}.json`), JSON.stringify(JSON.parse(stringify(data))))
```
Comment (AyAyEm) [inline build/build.js:103]: Could you add a comment explaining why it needs to double parse it?

File: build/stringify.js [hunk 6]
```diff
@@ -9,7 +9,7 @@ const isPrimitive = obj => obj === null || ['string', 'number', 'boolean'].inclu
 const isArrayOfPrimitive = obj => Array.isArray(obj) && obj.every(isPrimitive)
 const format = arr => `^^^[ ${arr.map(val => JSON.stringify(val)).join(', ')} ]`
 const replacer = (key, value) => isArrayOfPrimitive(value) ? format(value) : value // objectsort(value)
-const expand = str => str.replace(/"\^\^\^(\[ .* \])"/g, (match, a) => a.replace(/\\"/g, '"')).replace(/\\\\"/g, "'")
-const stringify = obj => expand(JSON.stringify(Array.isArray(obj) ? dedupe(obj) : obj, replacer, 2))
+const expand = str => str.replace(/"\^\^\^(\[ .* ])"/g, (match, a) => a.replace(/\\"/g, '"')).replace(/\\\\"/g, "'")
+const stringify = (obj) => expand(JSON.stringify(Array.isArray(obj) ? dedupe(obj) : obj, replacer, 2))
```
Comment (AyAyEm) [inline build/stringify.js:13]: I think it's better to remove the parentheses to keep the project consistent
```suggestion
const stringify = obj => expand(JSON.stringify(Array.isArray(obj) ? dedupe(obj) : obj, replacer, 2))
```

**Review by AyAyEm — CHANGES_REQUESTED** [review]

**Comment by AyAyEm** [comment]
Also would be possible to edit the description for this pr? Just for keeping things clean.

File: build/stringify.js (hunk 6 above)
Comment (TobiTenno) [reply build/stringify.js:13]: I'm planning a styling update pass before I do the ES upgrade. Can I do it then?

**Comment by TobiTenno** [comment]
> Also could you edit the description for this pr? Just for keeping things clean.


To include or remove what, specifically?

**Comment by AyAyEm** [comment]
> To include or remove what, specifically?

We also minfied the json data, changed from fix to feat to perf and refactored two action scripts. But just adding that we minfied the data would be fine.

File: build/stringify.js (hunk 6 above)
Comment (AyAyEm) [reply build/stringify.js:13]: Sure if you plan to change it anyway it won't be a problem.

**Pull request merged**